module/surveys/helper.js

```diff
@@ -1036,6 +1052,24 @@ module.exports = class SurveysHelper {
                     submissionDocumentEvidences = surveySubmissionDocument[0].evidences;
 
                 } else {
+                    // join survey's program. PII data consent is given via this api call.
+                    if ( solutionDocument.programId && ( appVersion === "" || appVersion < 5.2 ) && userToken !== "" && Object.keys(roleInformation).length > 0 ) {
```

Comment:
@VISHNUDAS-tunerlabs  when roleInformation is not passed then ?

Reply:
and programJoined check status

Reply:
@ankitpws pvt program details need not be stored, so all cases we will get roleInformation